src/archive/fields.rs

```diff
@@ -0,0 +1,259 @@
+            return true;
+        }
+        if self.args {
+            return true;
+        }
+        if self.success {
+            return true;
+        }
+        if self.parent {
+            return true;
+        }
+        false
+    }
+
+    pub fn selected_fields(&self) -> Vec<String> {
+        let mut fields = vec![];
+        if self._all {
+            fields.push("name".to_string());
+            fields.push("args".to_string());
+            fields.push("success".to_string());
+            fields.push("parent_id".to_string());
+        } else {
+            if self.name {
+                fields.push("name".to_string());
+            }
+            if self.args {
+                fields.push("args".to_string());
+            }
+            if self.success {
+                fields.push("success".to_string());
+            }
+            if self.parent {
+                fields.push("parent_id".to_string());
+            }
+        }
+        fields
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct CallFields {
+    pub _all: bool,
+    pub success: bool,
+    pub name: bool,
+    pub args: bool,
+    pub parent: ParentCallFields,
+}
+
+impl CallFields {
+    pub fn new(value: bool) -> Self {
+        CallFields {
+            _all: value,
+            success: value,
+            name: value,
+            args: value,
+            parent: ParentCallFields::new(value),
+        }
+    }
+
+    pub fn any(&self) -> bool {
+        if self._all {
+            return true;
+        }
+        if self.success {
+            return true;
+        }
+        if self.name {
+            return true;
+        }
+        if self.args {
+            return true;
+        }
+        if self.parent.any() {
+            return true;
+        }
+        false
+    }
+
+    pub fn selected_fields(&self) -> Vec<&str> {
+        let mut fields = vec![];
+        if self._all {
+            fields.push("success");
+            fields.push("name");
+            fields.push("args");
+            fields.push("parent_id");
+        } else {
+            if self.success {
+                fields.push("success");
+            }
+            if self.name {
+                fields.push("name");
+            }
+            if self.args {
+                fields.push("args");
+            }
+            if self.parent.any() {
+                fields.push("parent_id");
+            }
+        }
+        fields
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct ExtrinsicFields {
+    pub _all: bool,
+    pub index_in_block: bool,
+    pub signature: bool,
+    pub success: bool,
+    pub hash: bool,
+    pub call: CallFields,
+}
+
+impl ExtrinsicFields {
+    pub fn new(value: bool) -> Self {
+        ExtrinsicFields {
+            _all: value,
+            index_in_block: value,
+            signature: value,
+            success: value,
+            hash: value,
+            call: CallFields::new(value),
+        }
+    }
+
+    pub fn any(&self) -> bool {
```

Comment:
this might be better implemented like
```
return self.a || self.b || self.c
```

Reply:
yea
your suggestion looks cleaner, will change